GitUpKit/Extensions/GCRepository+Utilities.m

```diff
@@ -205,11 +205,19 @@ - (BOOL)syncIndexWithWorkingDirectory:(NSError**)error {
         if (delta.submodule) {
           GCSubmodule* submodule = [self lookupSubmoduleWithName:delta.canonicalPath error:error];
           if (!submodule || ![self addSubmoduleToRepositoryIndex:submodule error:error]) {
-            return NO;
+            if (![[*error localizedDescription] hasSuffix:@"' has not been added yet"]) {
```

Comment:
Also, is it possible error is NULL here? If I remember my C-isms well enough, dereferencing error would crash, no?

Reply:
Good catch. I fixed both instances of that bug.

Of note, my fix is a bit too simple: it effectively disables this PR's bugfix, when no error is passed. But, _that_ new bug is niche enough that it should be OK; handling it right would mean more code, and some pointer juggling I'm not completely comfortable with.